code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -58,106 +58,91 @@
 	if(announcing_vox > world.time)
 		to_chat(src, "<span class='notice'>Please wait [DisplayTimeText(announcing_vox - world.time)].</span>")
 		return
-
-	var/message = input(src, "WARNING: Misuse of this verb can result in you being job banned. More help is available in 'Announcement Help'", "Announcement", src.last_announcement) as text|null
+	var/list/banned_characters = CONFIG_GET(keyed_list/vox_voice_blacklist)
+	var/list/usable_characters = GLOB.available_vox_voices.Copy()
+	for(var/B in banned_characters)
+		usable_characters -= GLOB.vox_config_translator[B]
+	var/character_to_use = input(src, "Choose what 15.ai character to use:", "15.ai Character Choice")  as null|anything in usable_characters
+	if(!character_to_use)
+		return
+	var/max_characters = 200 // magic number but its the cap 15 allows
+	var/message = input(src, "Use the power of 15.ai to say anything! (200 character maximum)", "15.ai VOX System", src.last_announcement) as text|null
 
 	if(!message || announcing_vox > world.time)
 		return
 
-	last_announcement = message
-
 	if(incapacitated())
 		return
 
 	if(control_disabled)
 		to_chat(src, "<span class='warning'>Wireless interface disabled, unable to interact with announcement PA.</span>")
 		return
 
-	var/list/words = splittext(trim(message), " ")
-	var/list/incorrect_words = list()
-
-	if(words.len > 30)
-		words.len = 30
-
-	for(var/word in words)
-		word = lowertext(trim(word))
-		if(!word)
-			words -= word
-			continue
-		if(!GLOB.vox_sounds[word])
-			incorrect_words += word
-
-	if(incorrect_words.len)
-		to_chat(src, "<span class='notice'>These words are not available on the announcement system: [english_list(incorrect_words)].</span>")
+	if(length(message) > max_characters)
+		to_chat(src, "<span class='notice'>You have too many characters! You used [length(message)] characters, you need to lower this to [max_characters] or lower.</span>")
+		return
+	var/regex/check_for_bad_chars = regex("\[^a-zA-Z!?.,' :\]+")
+	if(check_for_bad_chars.Find(message))
+		to_chat(src, "<span class='notice'>These characters are not available on the 15.ai system: [english_list(check_for_bad_chars.group)].</span>")
 		return
 
-	announcing_vox = world.time + VOX_DELAY
-
-	log_game("[key_name(src)] made a vocal announcement with the following message: [message].")
-
-	for(var/word in words)
-		play_vox_word(word, src.z, null)
-
-
-/proc/play_vox_word(word, z_level, mob/only_listener)
-
-	word = lowertext(word)
+	var/emotion_to_use = "Neutral"
+	switch(character_to_use)
+		if("GLaDOS (Homicidal)")
+			emotion_to_use = "Homicidal"
+			character_to_use = "GLaDOS"
+		if("GLaDOS (Neutral)")
+			character_to_use = "GLaDOS"
+		if("Twilight Sparkle (Happy)")
+			emotion_to_use = "Happy"
+			character_to_use = "Twilight Sparkle"
+		if("Twilight Sparkle (Neutral)")
+			character_to_use = "Twilight Sparkle"
+	last_announcement = message
 
-	if(GLOB.vox_sounds[word])
+	announcing_vox = world.time + VOX_DELAY

```

Comment:
You probably never want there to be multiple requests at once. It makes sense to disable vox entirely as soon as a request gets sent and only start the cooldown after the sound gets played for the players.